lib/classifier-reborn/backends/bayes_memory_backend.rb

```diff
@@ -0,0 +1,72 @@
+# Author: Sawood Alam <@ibnesayeed>
+
+module ClassifierReborn
+  class BayesMemoryBackend
+    def initialize
+      @total_words     = 0
+      @total_trainings = 0
+      @category_counts = Hash.new { |h, k| h[k] = {training: 0, word: 0} }
+      @categories      = {}
+    end
+
+    def total_words
+      @total_words
+    end
+
+    def update_total_words(diff)
+      @total_words += diff
+    end
+
+    def total_trainings
+      @total_trainings
+    end
+
+    def update_total_trainings(diff)
+      @total_trainings += diff
+    end
+
+    def category_training_count(category)
+      @category_counts[category][:training]
+    end
+
+    def update_category_training_count(category, diff)
+      @category_counts[category][:training] += diff
+    end
+
+    def category_has_trainings?(category)
+      @category_counts.key?(category) && @category_counts[category][:training] > 0
+    end
+
+    def category_word_count(category)
+      @category_counts[category][:word]
+    end
+
+    def update_category_word_count(category, diff)
+      @category_counts[category][:word] += diff
+    end
+
+    def add_category(category)
+      @categories[category] ||= Hash.new(0)
+    end
+
+    def category_keys
```

Comment:
In the `bayes.rb` file, there is a method named `categories` that returns a more natural list of category names. However, in the data structure, those categories are converted to symbols (see CategoryNamer class: `name.to_s.tr('_', ' ').capitalize.intern`). I want to keep the transformations and logic away from the storage (backend) class. So, the `category_keys` method returns the form that was stored the data structure.